src/status_im/contexts/settings/wallet/data_store.cljs

```diff
@@ -0,0 +1,47 @@
+(ns status-im.contexts.settings.wallet.data-store)
+
+(defn extract-keypair-name
+  [db key-uids-set]
+  (when (= (count key-uids-set) 1)
+    (let [key-uid  (first key-uids-set)
+          keypairs (get-in db [:wallet :keypairs])]
+      (->> (filter #(= (:key-uid %) key-uid) keypairs)
+           first
+           :name))))
+
+(defn update-keypair
+  [keypairs key-uid update-fn]
+  (mapcat (fn [keypair]
+            (if (= (keypair :key-uid) key-uid)
+              (if-let [updated (update-fn keypair)]
+                [updated]
+                [])
+              [keypair]))
+   keypairs))
+
+(defn make-accounts-fully-operable
+  [accounts key-uids-set]
+  (reduce-kv
+   (fn [acc k account]
+     (if (and (contains? key-uids-set (:key-uid account))
+              (= (keyword (:operable account)) :no))
+       (assoc acc k (assoc account :operable :fully))
+       (assoc acc k account)))
+   {}
+   accounts))
+
+(defn- make-keypairs-accounts-fully-operable
+  [accounts]
+  (map (fn [account]
+         (assoc account :operable :fully))
+       accounts))
+
+(defn make-keypairs-fully-operable
```

Comment:
Defiantly, just i couldn't find any data-store-tests so though event-test here should be enough.

Reply:
@mohsen-ghafouri, using our own codebase as good examples for testing/quality isn't reliable. Some data store functions have tests, but most don't. That's also because it's legacy code and in the past there weren't many tests. It's more a thing of "where we want to be" because the present is not yet great.

To me it's simple, we should always **consider** testing non-trivial behavior from public functions, it doesn't matter where it's defined (sub handlers, event handlers, data stores, utils). It's something we should reach some consensus sooner rather than later. For someone doing TDD this recommendation is a given/automatic.

On the optimistic side, for subscription handlers we improved a lot in the last year. For event handlers it's starting to get traction.

Reply:
the most challenges for me at least is lack of good documentation about different kind of test and how we cover effects/sub/fun with test. that's the reason i always looking for some example in codebase. but if this is a direction we want to go first we should prepare an insightful documentation about tests with some examples.

Reply:
I agree. Our standards for testing should be clearer because we can't assume we are all magically on the same page. Thanks for the feedback :)

Reply:
I always welcome feedback as it helps increase the quality of the code. I believe that learning and improving the codebase happens during code reviews, and I’m happy to be part of that process.

Thank you @ilmotta for all your feedbacks.